File: 7/Java/Solution.java
public class Solution {
    public int find(int m, int[] arr) {
        int i = 0;
        int j = arr.length - 1;
        int res = 0;
        while (i < j) {
            int sum = arr[i] + arr[j];
            if (sum > m) {
                j--;
            } else if (sum < m) {
                i++;
            } else {
                res++;
                i++;
                j--;
            }
        }
        return res;
    }

    public static void main(String[] args) {

        int[] arr = {1, 3, 5, 6, 8, 13, 16};
        
        System.out.println(new Solution().find(14, arr));
    }
}

File: 7/Python/Solution.py
# -*- coding:utf-8 -*-


class Solution:
    def Find(self, m, array):
        i = 0
        j = len(array) - 1
        res = 0
        while i < j:
            sum = array[i] + array[j]
            if sum > m:
                j -= 1
            elif sum < m:
                i += 1
            else:
                res += 1
                j -= 1
                i += 1

        return res


print(Solution().Find(14, [1,3,5,6,8,13,16]))
